pyccel/parser/syntax/headers.py

```diff
@@ -167,22 +174,26 @@ def __init__(self, **kwargs):
 
         dtype: list fo str
         """
-        self.dtype = kwargs.pop('dtype')
+        self.dtypes = kwargs.pop('dtypes')
         self.const = kwargs.pop('const')
 
         super(UnionTypeStmt, self).__init__(**kwargs)
 
     @property
     def expr(self):
-        l = [i.expr for i in self.dtype]
+        dtypes = [i.expr for i in self.dtypes]
         if self.const:
-            for e in l:
-                e["is_const"] = True
-
-        if len(l)>1:
-            return UnionType(l)
-        else:
-            return l[0]
+            for d_type in dtypes:
+                d_type["is_const"] = True
+        if len(dtypes)==1:
+            return dtypes[0]
+        if any(isinstance(d_type, FuncType) for d_type in self.dtypes):
+            msg = 'Functions in a uniontype are not supported yet'
+            errors.report(msg,
+                        severity='fatal')
```

Comment:
Can you add a test in tests/errors to make sure that this is correctly raised please

Reply:
Ok I will add some tests